pkg/cri/server/image_pull.go

```diff
@@ -786,3 +795,29 @@ func (rt *pullRequestReporterRoundTripper) RoundTrip(req *http.Request) (*http.R
 	}
 	return resp, err
 }
+
+// Given that runtime information is not passed from PullImageRequest, we depend on an experimental annotation
+// passed from pod sandbox config to get the runtimeHandler. The annotation key is specified in configuration.
+// Once we know the runtime, try to override default snapshotter if it is set for this runtime.
+// See https://github.com/containerd/containerd/issues/6657
+func (c *criService) snapshotterFromPodSandboxConfig(ctx context.Context, imageRef string,
+	s *runtime.PodSandboxConfig) (string, error) {
+	snapshotter := c.config.ContainerdConfig.Snapshotter
+	if s == nil || s.Annotations == nil {
+		return snapshotter, nil
+	}
+
+	runtimeHandler, ok := s.Annotations[annotations.RuntimeHandler]
+	if !ok {
+		return snapshotter, nil
+	}
+
+	ociRuntime, err := c.getSandboxRuntime(s, runtimeHandler)
+	if err != nil {
+		return "", errors.Wrapf(err, "experimental: failed to get sandbox runtime for %s", runtimeHandler)
```

Comment:
use `fmt.Errorf` here? since we just deprecated pkg/error package.

Reply:
Done, remove using pkg/errors for consistency. Curious, though, why do we want to deprecate the pkg? The wrap and stacktrace is a pretty nice feature for investigation and debugging.

Reply:
I think we don't use stack helper in containerd and pkg/errors has been archived😂😂😂